lib/agent.js

```diff
@@ -72,6 +83,7 @@ class Agent {
   }
 }
 
+Agent.prototype.get = util.isWeakRefsSupported ? getFromWeakCache : getFromCache
```

Comment:
I think we can skip the weak cache version if this works?

Reply:
Sounds good, I'll leave the logic for now but use only the version without the weakCache and see what happens.

Reply:
It looks like it works, so I think we can switch to a version without `WeakRefs` easily.